napalm/eos.py

```diff
@@ -237,48 +237,94 @@ def get_interfaces_counters(self):
         return interface_counters
 
 
+    def get_bgp_neighbors(self):
+
+        commands = list()
+        commands.append('show ip bgp summary vrf all')
+        commands.append('show ip bgp neighbors vrf all | i remote AS|remote router ID|Description|^ *IPv[4-6] Unicast:')
+        commands.append('show ipv6 bgp summary vrf all')
+        commands.append('show ipv6 bgp neighbors vrf all | i remote AS|remote router ID|Description|^ *IPv[4-6] Unicast:')
+
+        output = self.device.run_commands(commands)
+
+        ##########################################
+        # no JSON available for show ip bgp neigh
+        # Using 'show ipv[4-6] bgp neighbors vrf all | i remote AS|remote router ID|Description|^ *IPv[4-6] Unicast:'
+        # NOTE: if there is no description, EOS does not print the line.
+
+
+        # Regex the output from show ip bgp neigh
+        def get_bgp_neighbor(needed_peer, vrf, output_to_parse):
+            import re
+
+            bgp_neighbors = dict()
+            bgp_peer = dict()
+            neighbor_regexp = re.compile('BGP neighbor is (.*?),')
+            description_regexp = re.compile('Description: (.*?)$')
+            remote_id_regexp = re.compile('remote router ID (.*?),')
+            vrf_regexp = re.compile('VRF (.*?)$')
+            IPv4_sent_regexp = re.compile('IPv4 Unicast: ( *)(\d*) ')
+            IPv6_sent_regexp = re.compile('IPv6 Unicast: ( *)(\d*) ')
+
+            for line in output_to_parse.splitlines():
+                if re.search(neighbor_regexp, line):
+                    peer = re.search(neighbor_regexp, line).group(1)
```

Comment:
As discussed, the regex is executed twice, so the if statement checks if it finds something or not